scapy/packet.py

```diff
@@ -638,6 +638,24 @@ def copy_fields_dict(self, fields):
         return {fname: self.copy_field_value(fname, fval)
                 for fname, fval in six.iteritems(fields)}
 
+    def _raw_packet_cache_field_value(self, fld, val, copy=False):
+        # type: (AnyField, Any, bool) -> Optional[Any]
+        """Get a value representative of a mutable field to detect changes"""
+        if fld.holds_packets:
+            # avoid copying packets (perf: #GH3894)
+            if fld.islist:
+                return [
+                    x.raw_packet_cache for x in val
```

Comment:
Looking at https://github.com/secdev/scapy/commit/766b99b6c9ed1c9122643ef5e7da51486dacc6af I'm not sure `raw_packet_cache` can be used here instead. I think (in its current form) the whole cache should be invalidated to avoid copying packets.